### ISSUE TITLE
qDebug/qWarning redirection to golang log, Qt_qInstallMessageHandler()

### ISSUE THREAD

**Issue by yshurik** [therecipe/qt#680]
Hi,

Really need redirection of qt messages to go logging system -  especially for MXE builds where qDebug/qWarning/etc are not shown at all - not possible to debug qml apps which do such logging.

I have written small module to do it in my app: https://github.com/yshurik/goqdebug
but wondering if this code can be integrated here - do you have some hint where to do this as core.h/cpp/go are generated and not a place to modify?

Some core.Qt_qInstallMessageHandler() call?

**Comment by therecipe** [comment]
Hey

Yeah, all the code is generated atm and I would rather not want to mix it with handwritten code, but I will think about it.

In the meantime you could simply make a go package out of it with:
```patch
diff --git a/qdebug.go b/qdebug.go
index ed21954..53e9ffb 100644
--- a/qdebug.go
+++ b/qdebug.go
@@ -1,4 +1,4 @@
-package main
+package goqdebug
 
 /*
 #include <stdint.h>
@@ -49,6 +49,6 @@ func goqdebug_GoMessageHandler(cmsg C.struct_QtCore_PackedString, msg_type int)
 	log.Println(msg_mark, msg)
 }
 
-func main() {
-	C.goqdebug_test()
-}
+//func main() {
+//	C.goqdebug_test()
+//}
```
and then import it with `import _ "github.com/yshurik/goqdebug"` from your projects.

Also you can get the console output for windows builds with `QT_DEBUG_CONSOLE=true` since a few days.

**Comment by yshurik** [comment]
Yes, it is fine for my project & tweaking but may need to add a call to install custom message handler from go - to save qt logs to a file, send via udp, etc - just give a way to a user.

**Comment by angiglesias** [comment]
This feature would be really useful. @therecipe is there any chance to have this functionality included?
